# Issue 19553: Issue with convert maps for facade parents

archive/issues_019316.json:
```json
{
    "body": "Consider the following:\n\n```\nsage: class Foo(Parent):\n....:     def __init__(self):\n....:         Parent.__init__(self, facade=True, category=Sets())\n....:     def _element_constructor_(self, x):\n....:         return int(x)\nsage: F = Foo()\nsage: F(5)\n---------------------------------------------------------------------------\nTypeError                                 Traceback (most recent call last)\n...\nTypeError: Cannot convert int to sage.structure.element.Element\nsage: mor = F.convert_map_from(ZZ)\nsage: mor\nConversion map:\n  From: Integer Ring\n  To:   <class '__main__.Foo_with_category'>\nsage: mor(5)\n---------------------------------------------------------------------------\nTypeError                                 Traceback (most recent call last)\n...\nTypeError: Cannot convert int to sage.structure.element.Element\n```\nwhich comes from the coercion framework creating a `DefaultConvertMap_unique` whose `_call_` method returns an `int`, since `S` is a facade for `int`'s, which cannot be converted to an `Element`.\n\n**Assignee:** @tscrim\n\n**CC:**  nils @nbruin SImonKing @jdemeyer @vbraun\n\nIssue created by migration from https://trac.sagemath.org/ticket/19553\n\n",
    "created_at": "2015-11-08T22:01:06Z",
    "labels": [
        "component: coercion",
        "bug"
    ],
    "milestone": "https://github.com/sagemath/sagetest/milestones/sage-6.10",
    "title": "Issue with convert maps for facade parents",
    "type": "issue",
    "url": "https://github.com/sagemath/sagetest/issues/19553",
    "user": "https://github.com/tscrim"
}
```
Consider the following:

```
sage: class Foo(Parent):
....:     def __init__(self):
....:         Parent.__init__(self, facade=True, category=Sets())
....:     def _element_constructor_(self, x):
....:         return int(x)
sage: F = Foo()
sage: F(5)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
...
TypeError: Cannot convert int to sage.structure.element.Element
sage: mor = F.convert_map_from(ZZ)
sage: mor
Conversion map:
  From: Integer Ring
  To:   <class '__main__.Foo_with_category'>
sage: mor(5)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
...
TypeError: Cannot convert int to sage.structure.element.Element
```
which comes from the coercion framework creating a `DefaultConvertMap_unique` whose `_call_` method returns an `int`, since `S` is a facade for `int`'s, which cannot be converted to an `Element`.

**Assignee:** @tscrim

**CC:**  nils @nbruin SImonKing @jdemeyer @vbraun

Issue created by migration from https://trac.sagemath.org/ticket/19553


